WebDevStudios/nextjs-wordpress-starter
# Feature/45 add checkbox field

Part Of https://github.com/WebDevStudios/nextjs-wordpress-starter/issues/45

### Link

Link to the branch preview generated by Vercel.

### Description

This feature adds "Checkbox Field" support for GravityForms. Along with this feature it includes the following

-`GfField.Checkbox` form field
- `Input.CheckboxGroup` component for displaying multiple checkboxes
- `Input.Checkbox` component for displaying a single checkbox
- Convert `fieldId` prop on `Input.Text` to `id` since `fieldId` is specific to GravityForms
- Add `Checkbox` field case to GravityForm `Fields`
- Add `ArraySchemaFactory` class based for handling Yup Array  schema
- Add ` getMinLengthSchema` method to `ArraySchemaFactory` for Checkbox validation

### Screenshot
Checkbox fields being selected
<img width="526" alt="Screen Shot 2021-01-12 at 12 00 18 PM" src="https://user-images.githubusercontent.com/25035900/104346799-d5630f80-54cd-11eb-8ea5-701b04f6f1fb.png">

Checkbox field data in form submit
<img width="442" alt="Screen Shot 2021-01-12 at 12 00 41 PM" src="https://user-images.githubusercontent.com/25035900/104346801-d5fba600-54cd-11eb-8dab-d1a48ff5aa8a.png">

### Verification

How will a stakeholder test this?

Note: Checkbox defaults are not working at this time. Required field validation is working only checks for one checkbox to be selected.

1. Checkout `test/45-add-checkbox-field` 
2. Make sure you're on the latest version of develop and and visit `http://localhost:3000/blog/form-checkbox-test`
2. Interact with checkbox fields from the test form
3. Submit and verify the form preview.

## Form Defaults Issue
While working on this feature, it became clear that the form defaults need enhanced. See https://github.com/WebDevStudios/nextjs-wordpress-starter/issues/86
